File: food_delivery_app/android-driver/app/src/main/java/com/smartsoko/driver/data/model/Driver.kt
package com.smartsoko.driver.data.model

import java.util.Date

enum class DriverStatus {
    OFFLINE, ONLINE, BUSY
}

data class Driver(
    val id: String = "",
    val ownerId: String = "",
    val name: String = "",
    val email: String = "",
    val phone: String = "",
    val vehicleType: String = "",
    val vehiclePlate: String = "",
    val isOnline: Boolean = false,
    val status: DriverStatus = DriverStatus.OFFLINE,
    val rating: Double = 0.0,
    val reviewCount: Int = 0,
    val currentLatitude: Double = 0.0,
    val currentLongitude: Double = 0.0,
    val createdAt: Date = Date(),
    val updatedAt: Date = Date()
)

File: temp_build_project/lib/domain/entities/driver.dart
class Driver {
  final String id;
  final String name;
  final String phone;
  final String? avatarUrl;
  final double rating;
  final int totalTrips;
  final String vehicleModel;
  final String vehicleColor;
  final String vehiclePlate;
  final String vehicleYear;
  final bool isAvailable;

  const Driver({
    required this.id,
    required this.name,
    required this.phone,
    this.avatarUrl,
    required this.rating,
    required this.totalTrips,
    required this.vehicleModel,
    required this.vehicleColor,
    required this.vehiclePlate,
    required this.vehicleYear,
    this.isAvailable = true,
  });
}
